crates/matrix-sdk-ui/src/notification_client.rs

```diff
@@ -63,10 +63,12 @@ impl NotificationClient {
 
         let mut timeline_event = room.event(event_id).await?;
 
-        if self.filter_by_push_rules
-            && !timeline_event.push_actions.iter().any(|a| a.should_notify())
-        {
-            return Ok(None);
+        if self.filter_by_push_rules {
+            if let Some(push_actions) = &timeline_event.push_actions {
```

Comment:
I think this might be more readable as a single if using `.as_ref().is_some_and(...)` for the push actions.